ElectronClient/gui/Header.jsx

```diff
@@ -83,14 +83,13 @@ class HeaderComponent extends React.Component {
 		}
 
 		if (this.props.zoomFactor !== prevProps.zoomFactor || this.props.size !== prevProps.size) {
-			const mediaQuery = window.matchMedia(`(max-width: ${550 * this.props.zoomFactor}px)`);
-			const showButtonLabels = !mediaQuery.matches;
+			this.determineButtonLabelState();
+		}
+	}
 
-			if (this.state.showButtonLabels !== showButtonLabels) {
-				this.setState({
-					showButtonLabels: !mediaQuery.matches,
-				});
-			}
+	componentDidMount() {
+		if (this.state.showButtonLabels == undefined) {
```

Comment:
Do we need this conditional? Is it just an optimisation? Ideally we'd run your update logic on every mount to ensure the state is correct.

Reply:
I guess it's kinda unnecessary. It's only relevant at the startup where `componentDidUpdate()` is already called, so running the `determineButtonLabelState()` again is unnecessary. 

If we add `showButtonLabels: true` to the state this is not required at all.
Thanks for the suggestion. I'll update it right away.